src/borg/helpers/parseformat.py

```diff
@@ -498,6 +498,7 @@ def validator(text):
 
 def archivename_validator():
     def validator(text):
+        text = replace_placeholders(text)
         if '/' in text or '::' in text or not text:
             raise argparse.ArgumentTypeError('Invalid repository name: "%s"' % text)
```

Comment:
unrelated to your change, but just noticed: the error message seems to be wrong, should say "Invalid archive name".